### DISCUSSION TITLE
How do i change conversion bitrate?

### DISCUSSION THREAD

**Discussion by StefanDorschu** [seanap/auto-m4b#22] [General]
Im new to docker, got it running but how do i change the bitrate of the conversion?

**Comment by seanap** [comment]
By default it should be grabbing the bitrate of the original book and automatically making the m4b conversion the exact same.  If you want to set it to a lower bitrate (32 or 64) then you can force it in the docker run command in the script.  I would avoid setting the bitrate to 128, because upscaling is just a waste of time/storage/energy for no benefit when starting with smaller bitrates. 

You'll need to [exec into the running docker container](https://github.com/seanap/auto-m4b#advanced-options) and modify [line 109](https://github.com/seanap/auto-m4b/blob/c1c48c64ce27101504c84cfa829edd2dee34b8ed/auto-m4b-tool.sh#L109) to something like `--audio-bitrate=64k`

**Reply by StefanDorschu** [reply]
Thank you, it worked. The files are in 320k MP3, so i needed to downscale :)

**Reply by seanap** [reply]
Ah, that makes a lot of sense, 320 is overkill for most audiobooks. Glad you got it working!

**Reply by seanap** [reply]
FYI this change will be overwritten if you update the docker, or if you remove and pull a new image.  Setting the bitrate hasn't been added to the environment variables but probably should be in a future release.

**Comment by StefanDorschu** [comment]
Maybe you can help me with beets-audible as well.

I think i configured everything as written, but i get the following errors:

Traceback (most recent call last):                                                                                                                                                                
  File "/usr/lib/python3.9/site-packages/beets/plugins.py", line 268, in load_plugins                                                                                                             
    namespace = __import__(modname, None, None)                                                                                                                                                   
ModuleNotFoundError: No module named 'beetsplug.copyartifacts'                                                                                                                                    
                                                                                                                                                                                                  
** error loading plugin audible:                                                                                                                                                                  
Traceback (most recent call last):                                                                                                                                                                
  File "/usr/lib/python3.9/site-packages/beets/plugins.py", line 268, in load_plugins                                                                                                             
    namespace = __import__(modname, None, None)                                                                                                                                                   
ModuleNotFoundError: No module named 'beetsplug.audible‘   




Any ideas what is did wrong?

Best regards.

Stefan

> Am 04.01.2023 um 18:37 schrieb seanap ***@***.***>:
> 
> 
> FYI this change will be overwritten if you update the docker, or if you remove and pull a new image. This hasn't been added to the config file or environment variables but probably should be.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/seanap/auto-m4b/discussions/22#discussioncomment-4594125>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AF3WKZ5OOA6W7J34CVJ73HTWQWYMTANCNFSM6AAAAAATQ5AVW4>.
> You are receiving this because you authored the thread.
>

**Reply by seanap** [reply]
Looks like an installation error, did you clone the github repo?

**Comment by StefanDorschu** [comment]
I copied the compose and config yaml and used docker-compose.

**Reply by seanap** [reply]
Thats why, remove what you've done and try following the instructions exactly.  It's a little different of an install compared to normal dockers, there are external files (included in the git clone) that need to be imported into the beets docker.  I thought about combining them all into my own docker but I don't want to maintain the version control of both beets and the audible plug-in.

**Comment by StefanDorschu** [comment]
I think that did it :)

Is there a way to rename the chapters in m4b-tool like


1. original chapter name —> chapter 1
2. original chapter name —> chapter 2

Just a simple naming from chapter 1 to chapter n.

Best regards,

Stefan

**Reply by seanap** [reply]
Not that I'm aware of, but it might be possible.  There are a couple ways I do this, using the auto-m4b follow the section on [manually editing](https://github.com/seanap/auto-m4b#to-manually-set-chapters) the chapters.txt file.  

The other way uses mp3tag, I wrote a simple action script that changes the TITLE tag to Chapter x
![image](https://user-images.githubusercontent.com/17012946/210655075-14913067-2efa-48b7-ae35-f34df55efb15.png)

You'll have to open the chapters as individual files in the settings:
![image](https://user-images.githubusercontent.com/17012946/210655414-f388d2d1-a473-48d2-b6d4-0fcd99505a25.png)
